### PR TITLE
Refactor build script

### PR DESCRIPTION
**Pull request by neomatrix369**
Note: please do not merge this PR till previous PR (https://github.com/AdoptOpenJDK/openjdk-build/pull/36) has been merged. Once its are merge git will show the actual change (as commits).

- Refactored blocks of code in build.sh into respective functions. 
- Made function name consistent across scripts in one case.

### CODE REVIEW

**Comment by neomatrix369** [comment]
Kicked off build job on jenkins - https://ci.adoptopenjdk.net/job/openjdk_build_x86-64_linux_mani_test/5/console, lets wait and see how far it goes.

**Comment by neomatrix369** [comment]
Running tests successfully:
```
-------------------------------------------------
TEST: java/lang/ClassLoader/deadlock/TestOneWayDelegate.sh
  shell: 4.726 seconds
TEST RESULT: Passed. Execution successful
--------------------------------------------------
TEST: java/lang/Double/ToHexString.java
  build: 0.252 seconds
  compile: 0.252 seconds
  main: 0.155 seconds
TEST RESULT: Passed. Execution successful
--------------------------------------------------
TEST: java/lang/Double/ToString.java
  build: 0.022 seconds
  compile: 0.022 seconds
  main: 0.108 seconds
```

**Comment by karianna** [comment]
I expect the diff of this PR to be small once the other one has been merged and this PR is rebased on that, so will review when that's done.

**Comment by neomatrix369** [comment]
@karianna its ready for a review

File: sbin/build.sh [hunk 1]
```diff
@@ -22,211 +22,260 @@ OPENJDK_REPO_NAME=$3
...
-  echo "Skipping ALSA download"
-else
-  wget -nc ftp://ftp.alsa-project.org/pub/lib/alsa-lib-1.0.27.2.tar.bz2
-  tar xvf alsa-lib-1.0.27.2.tar.bz2
-  rm alsa-lib-1.0.27.2.tar.bz2
-fi
+  echo "Checking for ALSA"
 
-echo "Checking for freetype"
+  FOUND_ALSA=$(find "$WORKING_DIR" -name "alsa-lib-1.0.27.2")
```
Comment (karianna) [inline sbin/build.sh]: Add a comment that this should we have this version up in it's own external version file - this is part of the versions auditing we'll be having to do later

Comment (karianna) [inline sbin/build.sh]: Same goes for the other versioned dependencies

File: sbin/build.sh [hunk 2]
```diff
@@ -22,211 +22,260 @@ OPENJDK_REPO_NAME=$3
...
+    # We get the files we need at $WORKING_DIR/installedfreetype
+    bash ./configure --prefix=$WORKING_DIR/$OPENJDK_REPO_NAME/installedfreetype $PARAMS && make all && make install
+
+    if [ $? -ne 0 ]; then
+      echo "${error}Failed to configure and build libfreetype, exiting"
+      exit;
+    else
+      echo "${good}Successfully configured OpenJDK with the FreeType library (libfreetype)!"
+    fi
+    echo $normal
```
Comment (karianna) [inline sbin/build.sh]: Missing `{}`

Comment (karianna) [inline sbin/build.sh]: In a few other places in this script as well.

File: sbin/build.sh [hunk 3]
```diff
@@ -22,211 +22,260 @@ OPENJDK_REPO_NAME=$3
...
+  cd $WORKING_DIR
+
+  echo "Retrieving cacerts file"
+
+  # Ensure it's the latest we pull in
+  rm -rf $WORKING_DIR/cacerts_area
+
+  git clone https://github.com/AdoptOpenJDK/openjdk-build.git cacerts_area
+  echo "cacerts should be here..."
+  file $WORKING_DIR/cacerts_area/security/cacerts
```
Comment (karianna) [inline sbin/build.sh]: Where is _here_?

File: sbin/build.sh [hunk 4]
```diff
@@ -22,211 +22,260 @@ OPENJDK_REPO_NAME=$3
...
+  checkingAndDownloadCaCerts
+}
+
+
+configuringBootJDKConfigureParameter()
+{
+  if [ -z "$JDK_BOOT_DIR" ] ; then
+    echo "JDK_BOOT_DIR is $JDK_BOOT_DIR"
+    JDK_BOOT_DIR=/usr/lib/java-1.7.0
+  else
```
Comment (karianna) [inline sbin/build.sh]: Version should be externalised / parameterized for non java 8 builds

Comment (neomatrix369) [inline sbin/build.sh]: `JDK_BOOT_DIR` can already be set via environment variable so this is fine here i think

File: sbin/build.sh (hunk 1 above)
Comment (neomatrix369) [inline sbin/build.sh]: Wasn't the primary purpose when I was extracting the code blocks, I can extract all these kinds of dependencies in an PR

File: sbin/build.sh (hunk 2 above)
Comment (neomatrix369) [inline sbin/build.sh]: Same applies here, could add the braces for consistency

File: sbin/build.sh (hunk 3 above)
Comment (neomatrix369) [inline sbin/build.sh]: Was part of the script already, @gdams what did you want to communicate with this echo?

File: sbin/build.sh (hunk 2 above)
Comment (neomatrix369) [inline sbin/build.sh]: Done via 3df715e2bfdddb9775caa46e6920632c173b8736

**Comment by karianna** [comment]
Good enough for me, small tidying can be done next PR, but this is a good step forwards

**Pull request merged**